core/src/test/java/io/questdb/test/griffin/engine/functions/groupby/FirstBooleanGroupByFunctionFactoryTest.java

```diff
@@ -0,0 +1,106 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2023 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.test.griffin.engine.functions.groupby;
+
+import io.questdb.cairo.TableWriter;
+import io.questdb.cairo.sql.Record;
+import io.questdb.cairo.sql.RecordCursor;
+import io.questdb.cairo.sql.RecordCursorFactory;
+import io.questdb.griffin.SqlException;
+import io.questdb.test.AbstractCairoTest;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class FirstBooleanGroupByFunctionFactoryTest extends AbstractCairoTest {
+
+    @Test
+    public void testAllNull() throws SqlException {
+
+        ddl("create table tab (f boolean)");
+
+        try (TableWriter w = getWriter("tab")) {
```

Comment:
nit: there is no need to deal with table writers and record cursor factories in these tests. You could simply use our utility methods like `assertQuery`:
https://github.com/questdb/questdb/blob/e4b6520e7b33187a0d752d53748aa7e981db3f81/core/src/test/java/io/questdb/test/griffin/engine/functions/groupby/FirstSymbolGroupByFunctionFactoryTest.java#L33-L45

Reply:
converted to assertQuery,

Reply:
Still keeping the `testAllNull` using TableWriter.
Boolean does not have a null.